src/components/nav-icon-item/styles.ts

```diff
@@ -0,0 +1,61 @@
+import { styled } from '@linaria/react'
+import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react'
+
+type CommonNavIconItemProps = {
+  /**
+   * NOTE: data attribute to indicate the state of the nav item
+   * by default the styles is already applied with pseudo-class selector
+   * but to make it more flexible, consumer could use this attribute to
+   * change the styles
```

Comment:
**question:** Why do we need to have this flexibility for consumers? I'm struggling to see the value here 🤔

Reply:
> Why do we need to have this flexibility for consumers? I'm struggling to see the value here

My concern is to keep the state for this component. For example, when a notification `NavIconItem` is clicked, it would open a new node (i.e., a dialog). Then, this data-state attribute could be utilized to make the `NavIconItem` active since the relevant component is in an open state

Reply:
Yes, it makes sense that we need to put the "active" state in the DOM via an attribute, but I don't understand why we also need to do that for hover or focus?

Reply:
I don't think we need that hover and focus states to be placed as part of `data-state` attribute 😶, just mark them with pseudo selector